plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -339,6 +346,87 @@ public void testOptimizingPartitionsOfV2TableWithGlobalEqualityDeleteFile()
                                 .toArray(String[]::new));
     }
 
+    @Test(timeOut = 90_000)
+    public void testOptimizeDuringWriteOperations()
+            throws Exception
+    {
+        int invocationCount = 3;
+        boolean optimizeSucceeded = false;
+        for (int attempt = 0; attempt < invocationCount; attempt++) {
+            optimizeSucceeded = runOptimizeDuringWriteOperations();
+            if (optimizeSucceeded) {
+                break;
+            }
+        }
+        assertTrue(optimizeSucceeded, "Expected optimize to succeed in at least one test attempt");
```

Comment:
The test may be flaky. It doesn't seem unlikely to have DELETEs win the race with OPTIMIZE on all 3 attempts.
It currently is flaky when run locally (see comment below)

perhaps we could make it more deterministic by slowing down the deletes.
what about using blackhole with data rate limiting to slow down the delete query (`page_processing_delay` table property)?

then maybe you can get rid of 3 attempts.